packages/squiggle-lang/src/reducer/context.ts

```diff
@@ -2,14 +2,14 @@ import { Env } from "../dist/env.js";
 import { Stack } from "./stack.js";
 import { FrameStack, topFrameName } from "./frameStack.js";
 import { ReducerFn, evaluate } from "./index.js";
-import { Lambda } from "./lambda.js";
+import { BaseLambda } from "./lambda.js";
 
 export type ReducerContext = Readonly<{
   stack: Stack;
   environment: Env;
   frameStack: FrameStack;
   evaluate: ReducerFn;
-  inFunction: Lambda | undefined;
+  inFunction: BaseLambda | undefined;
```

Comment:
Wrong type, I guess a consequence of VS Code refactorings; it's best if you don't export `BaseLambda` from `lambda.ts` at all.

Reply:
I tried that, but it broke the BaseLambda class. This is called within BaseLambda, before Lambda is defined. 

<img width="491" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/ff5c21f2-4b92-49e9-9fbe-4d320e3858df">